invesalius/gui/widgets/canvas_renderer.py

```diff
@@ -360,22 +437,24 @@ def OnPaint(self, evt, obj):
         self.modified = False
         self._drawn = False
 
-    def _follow_draw_list(self):
+    def _follow_draw_list(self) -> list:
         out = []
+
         def loop(node, layer):
             for child in node.children:
                 loop(child, layer + child.layer)
                 out.append((layer + child.layer, child))
 
         for element in self.draw_list:
             out.append((element.layer, element))
-            if hasattr(element, 'children'):
-                loop(element,element.layer)
+            if hasattr(element, "children"):
+                loop(element, element.layer)
 
         return out
 
-
-    def draw_element_to_array(self, elements, size=None, antialiasing=False, flip=True):
+    def draw_element_to_array(
```

Comment:
I think you can type element as [protocol](https://docs.python.org/3/library/typing.html#typing.Protocol) that implements `draw_to_canvas` method.

Reply:
size is tuple of integer (width and height)

Reply:
antialiasing and flip are bool

Reply:
is this correct for the `Element` Protocol?
```python
class Element(Protocol):
    def draw_to_canvas(self, gc: wx.GraphicsContext, canvas: CanvasRendererCTX) -> None: ...
```

Reply:
I think `CanvasElement` is better name. But yes